sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -621,6 +621,17 @@ abstract class TypedImperativeAggregate[T] extends ImperativeAggregate {
     buffer(mutableAggBufferOffset) = serialize(getBufferObject(buffer))
   }
 
+  /**
+   * In-place replaces SparkSQL internally supported underlying storage format (BinaryType),
+   * with the aggregation buffer object stored at buffer's index `mutableAggBufferOffset`.
+   *
+   * This is only called when AggregatingAccumulator running on driver, after the framework
+   * shuffle in aggregate buffers.
+   */
+  final def deserializeAggregateBufferInPlace(buffer: InternalRow): Unit = {
```

Comment:
can't we just inline it? It seems overkill to introduce a new API in the base class.